specification/keyvault/Security.KeyVault.RBAC/models.tsp

```diff
@@ -0,0 +1,462 @@
+
+  /**
+   * Decrypt using an HSM key.
+   */
+  DecryptHsmKey: "Microsoft.KeyVault/managedHsm/keys/decrypt/action",
+
+  /**
+   * Wrap using an HSM key.
+   */
+  WrapHsmKey: "Microsoft.KeyVault/managedHsm/keys/wrap/action",
+
+  /**
+   * Unwrap using an HSM key.
+   */
+  UnwrapHsmKey: "Microsoft.KeyVault/managedHsm/keys/unwrap/action",
+
+  /**
+   * Sign using an HSM key.
+   */
+  SignHsmKey: "Microsoft.KeyVault/managedHsm/keys/sign/action",
+
+  /**
+   * Verify using an HSM key.
+   */
+  VerifyHsmKey: "Microsoft.KeyVault/managedHsm/keys/verify/action",
+
+  /**
+   * Create an HSM key.
+   */
+  CreateHsmKey: "Microsoft.KeyVault/managedHsm/keys/create",
+
+  /**
+   * Delete an HSM key.
+   */
+  DeleteHsmKey: "Microsoft.KeyVault/managedHsm/keys/delete",
+
+  /**
+   * Export an HSM key.
+   */
+  ExportHsmKey: "Microsoft.KeyVault/managedHsm/keys/export/action",
+
+  /**
+   * Release an HSM key using Secure Key Release.
+   */
+  ReleaseKey: "Microsoft.KeyVault/managedHsm/keys/release/action",
+
+  /**
+   * Import an HSM key.
+   */
+  ImportHsmKey: "Microsoft.KeyVault/managedHsm/keys/import/action",
+
+  /**
+   * Purge a deleted HSM key.
+   */
+  PurgeDeletedHsmKey: "Microsoft.KeyVault/managedHsm/keys/deletedKeys/delete",
+
+  /**
+   * Download an HSM security domain.
+   */
+  DownloadHsmSecurityDomain: "Microsoft.KeyVault/managedHsm/securitydomain/download/action",
+
+  /**
+   * Check status of HSM security domain download.
+   */
+  DownloadHsmSecurityDomainStatus: "Microsoft.KeyVault/managedHsm/securitydomain/download/read",
+
+  /**
+   * Upload an HSM security domain.
+   */
+  UploadHsmSecurityDomain: "Microsoft.KeyVault/managedHsm/securitydomain/upload/action",
+
+  /**
+   * Check the status of the HSM security domain exchange file.
+   */
+  ReadHsmSecurityDomainStatus: "Microsoft.KeyVault/managedHsm/securitydomain/upload/read",
+
+  /**
+   * Download an HSM security domain transfer key.
+   */
+  ReadHsmSecurityDomainTransferKey: "Microsoft.KeyVault/managedHsm/securitydomain/transferkey/read",
+
+  /**
+   * Start an HSM backup.
+   */
+  StartHsmBackup: "Microsoft.KeyVault/managedHsm/backup/start/action",
+
+  /**
+   * Start an HSM restore.
+   */
+  StartHsmRestore: "Microsoft.KeyVault/managedHsm/restore/start/action",
+
+  /**
+   * Read an HSM backup status.
+   */
+  ReadHsmBackupStatus: "Microsoft.KeyVault/managedHsm/backup/status/action",
+
+  /**
+   * Read an HSM restore status.
+   */
+  ReadHsmRestoreStatus: "Microsoft.KeyVault/managedHsm/restore/status/action",
+
+  /**
+   * Generate random numbers.
+   */
+  RandomNumbersGenerate: "Microsoft.KeyVault/managedHsm/rng/action",
+}
+
+/**
+ * The role scope.
+ */
+union RoleScope {
+  string,
+
+  /**
+   * Global scope
+   */
+  Global: "/",
+
+  /**
+   * Keys scope
+   */
+  Keys: "/keys",
+}
+
+/**
+ * Role definition.
+ */
+model RoleDefinition {
+  /**
+   * The role definition ID.
+   */
+  @visibility("read")
+  id?: string;
```

Comment:
Under what circumstances is this not returned?

Reply:
My understanding is that in practice this is always returned. However, like with other KV resources, this has always been optional in the Swagger. We've talked to the service team about identifying which resource attributes could be marked as required, but if I recall correctly, we were told that this wouldn't happen. cc @heaths @sandeep-sen